cpp/src/io/json/write_json.cu

```diff
@@ -0,0 +1,733 @@
+    // (using null representation and delimiter):
+    //
+    return struct_to_strings(str_table_view,
+                             column_names_view,
+                             row_begin_wrap.value(stream_),
+                             row_end_wrap.value(stream_),
+                             column_seperator.value(stream_),
+                             value_seperator.value(stream_),
+                             narep,
+                             options_.is_enabled_include_nulls(),
+                             stream_,
+                             rmm::mr::get_current_device_resource());
+  }
+
+ private:
+  json_writer_options const& options_;
+  rmm::cuda_stream_view stream_;
+  rmm::mr::device_memory_resource* mr_;
+  string_scalar const column_seperator{":"};
+  string_scalar const value_seperator{","};
+  string_scalar const row_begin_wrap{"{"};
+  string_scalar const row_end_wrap{"}"};
+  string_scalar narep{options_.get_na_rep()};
+};
+
+}  // namespace
+
+std::unique_ptr<column> make_strings_column_from_host(host_span<std::string const> host_strings,
+                                                      rmm::cuda_stream_view stream)
+{
+  std::string const host_chars =
+    std::accumulate(host_strings.begin(), host_strings.end(), std::string(""));
+  auto d_chars = cudf::detail::make_device_uvector_async(host_chars, stream);
+  std::vector<cudf::size_type> offsets(host_strings.size() + 1, 0);
+  std::transform_inclusive_scan(host_strings.begin(),
+                                host_strings.end(),
+                                offsets.begin() + 1,
+                                std::plus<cudf::size_type>{},
+                                [](auto& str) { return str.size(); });
+  auto d_offsets = cudf::detail::make_device_uvector_sync(offsets, stream);
+  return cudf::make_strings_column(
+    host_strings.size(), std::move(d_offsets), std::move(d_chars), {}, 0);
+}
+
+std::unique_ptr<column> make_column_names_column(host_span<column_name_info const> column_names,
+                                                 size_type num_columns,
+                                                 rmm::cuda_stream_view stream)
+{
+  std::vector<std::string> escaped_column_names;
+  std::transform(column_names.begin(),
+                 column_names.end(),
+                 std::back_inserter(escaped_column_names),
+                 [](column_name_info const& name_info) {
+                   std::string name = name_info.name;
+                   char const quote = '\"';
+                   if (name.empty() ||           // no header name
+                       name.front() == quote) {  // header already quoted
+                     return name;
+                   }
+
+                   // escape any quotes
+                   size_t pos = 0;
+                   while ((pos = name.find(quote, pos)) != name.npos) {
+                     name.insert(pos, 1, quote);
```

Comment:
This looks incorrect to me.
Can you add some tests with embedded double-quote `"` characters to see if this is right?

Reply:
Great catch. This logic is for csv. I updated the code with JSON escape strings, and also handled it in json reader.